dask/dataframe/groupby.py

```diff
@@ -1044,17 +1144,13 @@ def nunique(self, split_every=None, split_out=1):
 
     @derived_from(pd.core.groupby.SeriesGroupBy)
     def aggregate(self, arg, split_every=None, split_out=1):
```

Comment:
The short circuit-behavior is no longer necessary, since this is addressed by the added code. For `sum, size, count` the old implementation should be faster. For `var`, `mean`, `std`, the new implementation should be faster (less conversions). In all cases, this removes one additional branch to keep track of. I'm happy to re-add the original code, if you'd prefer.